app/org/sagebionetworks/bridge/services/ConsentService.java

```diff
@@ -256,8 +259,12 @@ public void consentToResearch(Study study, SubpopulationGuid subpopGuid, StudyPa
         account.setSharingScope(SharingScope.NO_SHARING);
         accountDao.updateAccount(account, false);
 
-        sendWithdrawEmail(study, participant.getExternalId(), account, withdrawal, withdrewOn);
-
+        for (SubpopulationGuid subpopGuid : account.getAllConsentSignatureHistories().keySet()) {
+            Subpopulation subpop = subpopService.getSubpopulation(study.getStudyIdentifier(), subpopGuid);
+            if (!subpop.isAutoSendConsentSuppressed()) {
+                sendWithdrawEmail(study, participant.getExternalId(), account, withdrawal, withdrewOn);
+            }
+        }
```

Comment:
This used to only send one email. Now it sends an email for every subpop. This is probably not our intent.

Incidentally, we probably want a unit test case with multiple subpops. (Both send and don't send cases.)

Reply:
I felt like this should be the intended behavior, but now I'm pausing on the case where one consent is superseded by another one. It might be strange to get a withdrawal email for both if you reconnected to a new consent. Unfortunately there's no way to determine which consents still hold and which don't right now.

I can revert to sending one withdrawal, and then suppressing if *all* the consents suppress notification. In a study that has never relied on email for example.

Reply:
Added test of two consents, one of which suppresses email, and two where both suppress email. Behavior is as expected.

Reply:
I mistook the purpose of this email, it is actually only sent to the study consent administrator, not to the study participant. I will walk back the changes to withdrawal notifications... the only thing they need to do is not send if the study has no consent email set.